### PR TITLE
tests: Update tests to use Go Testing

### PR DESCRIPTION
**Pull request by snehachhabria**
**Description**:

Update the existing tests to use go testing and improve coverage to 80%

Part of #1704 and #1489

Signed-off-by: Sneha Chhabria <snchh@microsoft.com>

**Testing done**:
Unit tests updated 

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [X] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? `no
    -   Did you notify the maintainers and provide attribution? 

2. Is this a breaking change? `no`

### CODE REVIEW

**Review by trstringer — APPROVED** [review]

File: cmd/osm-controller/validate_test.go [hunk 1]
```diff
@@ -1,160 +1,149 @@
...
+		},
+		{
+			name:         "Cert Provider : InvalidProvider",
+			certProvider: "InvalidProvider",
+			expectError:  true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
```
Comment (nojnhuh) [inline cmd/osm-controller/validate_test.go:68]: Let's move the `assert := tassert.New(t)` from the top of the test to here inside the sub-test so if a test case fails it's easier to pinpoint which sub-test the error message refers to.

(Also applicable to `TestValidateCLIParams` below)

File: pkg/k8s/event_handlers_test.go [hunk 2]
```diff
@@ -21,65 +19,57 @@ const (
...
+	a.Nil(pubsubMsg.OldObj)
+
+	// Cast New obj, expect v1.Pod
+	podObj, castOk := pubsubMsg.NewObj.(*corev1.Pod)
+	a.True(castOk)
+	a.Equal(podObj.Name, "pod-name")
+	a.Equal(podObj.Namespace, testNamespace)
+
+	podAddChannel = events.Subscribe(announcements.PodAdded)
+	defer events.Unsub(podAddChannel)
```
Comment (nojnhuh) [inline pkg/k8s/event_handlers_test.go:60]: I can't imagine this is actually a problem, but merging the two `It`s into one function here means for `podAddChannel` the test was doing

```
Subscribe()
Unsub()
Subscribe()
Unsub()
```

where now it does

```
Subscribe()
Subscribe()
Unsub()
Unsub()
```

nbd if the test reliably passes, just an observation.

**Review by nojnhuh — APPROVED** [review]
Left one suggestion and one comment, neither should block this.

File: pkg/k8s/event_handlers_test.go (hunk 2 above)
Comment (shashankram) [reply pkg/k8s/event_handlers_test.go:60]: Let's not change the behavior of the existing test unless there's a bug.

Comment (snehachhabria) [reply pkg/k8s/event_handlers_test.go:60]: This test is primarily checking if correct events are being subscribed to and the assert statements are helping in verifying the same. The order of Subscribe() and Unsub() has changed but that doesn't impact behavior. We should be okay here

**Pull request merged**